src/io.jl

```diff
@@ -60,60 +74,137 @@ function create_internal_structures(connection)
 
     groups = [Group(row...) for row in TulipaIO.get_table(Val(:raw), connection, "groups_data")]
 
+    _query(table_name, col; where_pairs...) = begin
+        _q = "SELECT year, $col FROM $table_name"
+        if length(where_pairs) > 0
+            _q *=
+                " WHERE " *
+                join(("$k=$(TulipaIO.FmtSQL.fmt_quote(v))" for (k, v) in where_pairs), " AND ")
+        end
+        DuckDB.query(connection, _q)
+    end
+    function _get_stuff(table_name, col; where_pairs...)
+        result = _query(table_name, col; where_pairs...)
+        Dict(row.year => getproperty(row, Symbol(col)) for row in result)
+    end
+
```

Comment:
Do you have to/want to initialize all the years in the dict? Because I see in later codes, you always check whether if the key (year) exists or not, and if not, initialize with `get()`.

Reply:
This does not add all years because `result` is a query from `assets_data`. So only the years informed in that file are here.
So if you have `wind,2030`, `gas,2030` and `gas,2050`, your `result` for `wind` will have only 2030, while `gas` will have 2030 and 2050.